### PR TITLE
Issue unawaited promise error on symbol-less expressions

### PR DESCRIPTION
**Pull request by andrewbranch**
Makes #43593 less conservative. Previously, since we wanted to check whether the body of the conditional references the same thing that’s being tested for truthiness, we had to get a symbol for the `if` expression and find references to that same symbol. We can only find symbols on identifiers and access expressions, so we failed to come up with a suitable location on call expressions like the one in the blog post:

```ts
async function foo(): Promise<boolean> {
    return false;
}

async function bar(): Promise<string> {
    if (foo()) { // <- no error
        return "true";
    }
    return "false";
}
```

This PR adds the error back in cases like these where we can’t easily tell whether or not the expression is used again in the function body.

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -35720,13 +35719,13 @@ namespace ts {
                 return;
             }
 
-            const testedSymbol = getSymbolAtLocation(testedNode);
-            if (!testedSymbol) {
+            const testedSymbol = testedNode && getSymbolAtLocation(testedNode);
+            if (!testedSymbol && !isPromise) {
```
Comment (andrewbranch) [inline src/compiler/checker.ts:35723]: I don’t know that we actually need have divergent behavior for unawaited promises and uncalled expressions here—what do we expect for something like

```ts
const or = x => y => x || y;

if (or(true)) { // today: no error, but maybe should be?
  // ...
}
```

**Review by sandersn — APPROVED** [review]
Looks good, although you are right that maybe it should apply to uncalled functions too.

**Comment by andrewbranch** [comment]
@typescript-bot test this
@typescript-bot user test this
@typescript-bot run dt

**Comment by andrewbranch** [comment]
Test infrastructure seemed to be having an issue, so I’m not going to worry about aligning the call expression behavior for now, which would be a breaking change that we haven’t validated. We can revisit in the future but I’d probably lean toward waiting to hear user feedback.

**Pull request merged**